Assets/Scripts/Ability/AbilityManager.cs

```diff
@@ -0,0 +1,242 @@
+        {
+            if (castTarget == CastTarget.Position)
+            {
+                _slots[_slotNumber].transform.SetParent(null, true);
+                _slots[_slotNumber].transform.position = _position;
+                _slots[_slotNumber].Reinit();
+                _slots[_slotNumber].Play();
+            }
+            else if (castTarget != CastTarget.Projectile)
+            {
+                _slots[_slotNumber].transform.SetParent(_slotParent.transform);
+                _slots[_slotNumber].transform.localPosition = Vector3.zero;
+                _slots[_slotNumber].transform.rotation = _playerCharacterTransform.rotation;
+                _slots[_slotNumber].Reinit();
+                _slots[_slotNumber].Play();
+            }
+        }
+
+        private void ExecuteAbility()
+        {
+            var ability = _abilities[_slotNumber];
+            switch (ability.CastTarget)
+            {
+                case CastTarget.Self:
+                    ability.Cast(_playerCharacterTransform.position);
+                    break;
+                case CastTarget.Direction:
+                    ability.Cast(_playerCharacterTransform.position, _playerCharacterTransform.forward);
+                    break;
+                case CastTarget.Position:
+                    ability.Cast(_position);
+                    break;
+                case CastTarget.Projectile:
+                    ability.Cast(_playerCharacterTransform.position, _playerCharacterTransform.forward, _playerCharacterTransform.rotation);
+                    break;
+                default:
+                    throw new NotImplementedException($"{nameof(CastTarget)} enum value is not implemented!");
+            }
+        }
+
+        internal void RemoveEffect()
+        {
+            _slots[_slotNumber].transform.SetParent(null, true);
+        }
+
+        public void Setup(List<Ability> abilities)
+        {
+            _abilities = abilities;
+            SetVfx();
+        }
+
+        public void Destruct()
+        {
+            _abilities.Clear();
+        }
+
+        private void UseAbilityAim(int slot)
+        {
+            if (_playerInput.currentControlScheme == "Keyboard&Mouse")
+            {
+                var castTarget = _abilities[slot].CastTarget;
+                if (castTarget == CastTarget.Direction || castTarget == CastTarget.Projectile)
+                {
+                    Vector3 direction = _positionCast.GetCursorDirection(_cam, _layerMask, _playerCharacterTransform);
+                    _playerCharacterTransform.rotation = Quaternion.LookRotation(direction);
+                }
+                else if (castTarget == CastTarget.Position)
+                    _position = _positionCast.GetCursorPosition(_cam, _layerMask);
+            }
+        }
+
+        private bool IsValidCastTarget(CastTarget castTarget)
+        {
+            return castTarget == CastTarget.Self ||
```

Comment:
I feel like this could be done better.

Reply:
completely removed, since it is impossible to select anything other than a valid cast target